js/src/data/actions.js

```diff
@@ -929,3 +939,20 @@ export function* updateMCProductVisibility( ids, visible ) {
 		throw error;
 	}
 }
+
+/**
+ * Request a new review for the connected account
+ */
+export function* sendMCReviewRequest() {
+	try {
+		const response = yield apiFetch( {
+			path: `${ API_NAMESPACE }/mc/review`,
+			method: 'POST',
+		} );
+
+		return receiveMCReviewRequest( response );
```

Comment:
🚧 

The received `response` will not be passed to wp-data reducer, and it further causes the UI won't update after sending the review request.


https://user-images.githubusercontent.com/17420811/167812912-bd99ee96-36f9-4469-b62d-74750c9050a8.mp4



Since the `receiveMCReviewRequest` is also a generator function, it needs to be called with `yield` here, such as `return yield receiveMCReviewRequest( response );` . Or, change the `receiveMCReviewRequest` to a non-generator function.

Reply:
Thanks for catching this 🥇 I totally missed the yield